alarms.yaml

```diff
@@ -77,13 +77,32 @@ Resources:
       Dimensions:
       - Name: Cluster
         Value: !Ref ClusterName
-      EvaluationPeriods: 2
+      EvaluationPeriods: 15
```

Comment:
The fact that status is an int and these are using `ComparisonOperator: GreaterThanThreshold`, I think will mean that going from green to red status will trigger both ClusterStatusAlarm and RedClusterStatusAlarm. Is that a possible status transition? Or would it always go yellow first, green->yellow->red?

I'm wondering if changing `ClusterStatusAlarm` to a `YellowClusterStatusAlarm` would help us articulate more clearly what's happening. I'm looking at the doc and it seems  a little tricky  https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cw-alarm.html#cfn-cloudwatch-alarms-comparisonoperator

Reply:
Yeah, this change is a bit of a hack due to the (surprising!) lack of an 'in between' comparison operator.

> in some scenarios we will now receive a duplicate alarm (e.g. if the cluster is red for 15 minutes or more)

There are a couple of scenarios where'd get two alarms:

1. Cluster immediately goes from green into red status (or passes through yellow status very quickly) and _stays_ in red status for at least 15 minutes. In that case we'd get `RedClusterStatusAlarm` (after 1 minute) and _then_ `ClusterStatusAlarm` (after 15 minutes). 
2. Cluster goes into yellow status for 0-14 minutes and then moves into red status. In that case we'd get `RedClusterStatusAlarm` (after 1 minute of red status) and _then_ `ClusterStatusAlarm` (at the same time, or a few minutes later, depending on the length of yellow status that preceded the red status).

Although both of these cases are possible, they feel like relatively unusual scenarios, and if the cluster is red anyway a follow-up alarm doesn't seem like a huge problem to me.

I also considered renaming the original alarm to `YellowClusterStatusAlarm`, but since that doesn't accurately reflect all the scenarios that will trigger the alarm I decided to leave the slightly vaguer name!

I think that the alternatives to this hack would be:

1. Using [metric math to perform in between type logic](https://cupofcode.blog/aws-in-between-alarms/)
2. Post different CW metrics for yellow and red cluster status to allow for simpler, more precise alarms

But I wasn't convinced that either was worth the additional effort.

Reply:
Thanks for the explanation, you've clearly thought this through! I agree, I think this is an appropriate solution given our current setup. And if our cluster does go red, then 2 emails about it will be the least of our worries!